### PR TITLE
nrfx_glue: Add macros related to the pending status of IRQs

### PR DESCRIPTION
**Pull request by glennrub**

### CODE REVIEW

File: templates/nrfx_glue.h [hunk 1]
```diff
@@ -98,6 +98,19 @@ extern "C" {
  */
 #define NRFX_IRQ_DISABLE(irq_number)
 
+/**
+ * @brief Macro for setting a specific pending IRQ.
```
Comment (kl-cruz) [inline templates/nrfx_glue.h]: "Macro for setting a specific IRQ as pending." sounds better IMO

File: templates/nrfx_glue.h [hunk 2]
```diff
@@ -98,6 +98,19 @@ extern "C" {
...
+ *
+ * @param irq_number  IRQ number.
+ */
+#define NRFX_IRQ_PENDING_SET(irq_number)
+
+/**
+ * @brief Macro for clearing a specific pending IRQ.
```
Comment (kl-cruz) [inline templates/nrfx_glue.h]: Consider to change in the same way like in line 102

File: templates/nrfx_glue.h [hunk 3]
```diff
@@ -98,6 +98,19 @@ extern "C" {
...
 
+/**
+ * @brief Macro for setting a specific IRQ as pending.
+ *
+ * @param irq_number  IRQ number.
+ */
+#define NRFX_IRQ_PENDING_SET(irq_number)
+
+/**
+ * @brief Macro for clearing a specific IRQ from pending.
```
Comment (glennrub) [inline templates/nrfx_glue.h]: Does this sound ok? clear from pending?

Comment (anangl) [reply templates/nrfx_glue.h]: What about "clearing the pending status of a specific IRQ"?

Comment (glennrub) [reply templates/nrfx_glue.h]: Hence, aligning the 'set' also to: "setting the pending IRQ status of a specific IRQ"?

Comment (anangl) [reply templates/nrfx_glue.h]: Not necessarily. The description for `_SET` looks fine as it is now.

Comment (glennrub) [reply templates/nrfx_glue.h]: done.

Comment (anangl) [reply templates/nrfx_glue.h]: `NRFX_IRQ_PENDING_CLEAR` is not used in nrfx. Do you need it for a particular reason?
To be consistent, we should either not define this macro at all or define `NRFX_IRQ_IS_PENDING` as well to have a full set (but kind of forcing integrators to provide an implementation for unused macros).
I think I'd take this second path. @kl-cruz What you think?

Comment (glennrub) [reply templates/nrfx_glue.h]: I can remove the CLEAR, or add the IS_SET. I'm fine with both. In my use case i'm only using _SET.

**Comment by kl-cruz** [comment]
Last thing, could You add NRFX_IRQ_IS_PENDING function to cover all NVIC-irq-pending cases?

**Comment by glennrub** [comment]
Added IS_PENDING macro.

File: templates/nrfx_glue.h [hunk 4]
```diff
@@ -98,6 +98,27 @@ extern "C" {
...
+ * @brief Macro for clearing the pending status of a specific IRQ.
+ *
+ * @param irq_number  IRQ number.
+ */
+#define NRFX_IRQ_PENDING_CLEAR(irq_number)
+
+/**
+ * @brief Macro for checking the pending status of a specific IRQ.
+ *
+ * @retval true  If interrupt status is pending.
```
Comment (anangl) [inline templates/nrfx_glue.h]: If the IRQ is pending.

**Comment by glennrub** [comment]
Updated comment in templates/nrfx_glue.h and commit message according to suggestion.

**Comment by anangl** [comment]
Thanks.

**Pull request merged**